proposals/NNNN-isolated-any-functions.md

```diff
@@ -0,0 +1,516 @@
+function value:
+
+- If the function is dynamically non-isolated, the value of `isolation`
+  is `nil`.
+- If the function is dynamically isolated to a global actor type `G`,
+  the value of `isolation` is `G.shared`.
+- If the function is dynamically isolated to a specific actor reference,
+  the value of `isolation` is that actor reference.
+
+### Distributed actors
+
+Function values cannot generally be isolated to a distributed actor
+unless the actor is known to be local.  When a distributed actor *is*
+local, function values isolated to the actor can be converted to
+`@isolated(any)` type as above.  The `isolation` property presents
+the distributed actor as an `(any Actor)?` using the same mechanism
+as [`#isolation`](https://github.com/apple/swift-evolution/blob/main/proposals/0420-inheritance-of-actor-isolation.md#isolated-distributed-actors).
+
+### Isolation checking
+
+Since the isolation of a dynamically-isolated function value is
+statically unknown, calls to it typically cross an isolation barrier.
+This means that the call must be `await`ed even if the function is
+synchronous, and the arguments and result must satisfy the usual
+sendability restrictions for cross-isolation calls.
+
+If `f` is an immutable binding (such as a local `let` or a non-`inout`
+parameter) of `@isolated(any)` function type, then a call to `f` does
+not cross an isolation barrier if the current context is isolated
+to a derivation of the expression `f.isolation`.
+
+A context is isolated to a derivation of a expression `E` if it has an
+isolated capture that is initialized to a derivation of `E`.
+
+An expression is a derivation of `E` if:
+
+- it has the exact form of `E`;
+- it is a reference to a capture or immutable binding immediately
+  initialized with a derivation of `E`; or
+- it is a non-optional derivation of `E`.
+
+When analyzing an expression for these rules, certain non-instrumental
+differences in expression syntax and behavior must be ignored.  This
+and certain other terms in this section are intended to match the
+definitions laid out in [SE-0420](https://github.com/apple/swift-evolution/blob/main/proposals/0420-inheritance-of-actor-isolation.md#generalized-isolation-checking).
+
+For example:
+
+```swift
+func delay(operation: @isolated(any) () -> ()) {
+  let isolation = operation.isolation
+  Task { [isolated isolation] in
+    print("waking")
+    operation() // <-- does not cross an isolation barrier and so is synchronous
+    print("finished")
+  }
+}
+```
+
+TODO: Add some way to use `assumeIsolated` to assert that we're currently
+isolated to the isolation of an `@isolated(any)` function, such that
+calls within that context no longer cross isolation boundaries.
+Maybe this could also decompose the function value?
+
+### Adoption in task-creation routines
+
+There are a large number of functions in the standard library that create
+tasks:
+- `Task.init`
+- `Task.detached`
+- `TaskGroup.add`
```

Comment:
Similar to the other proposal I wrote I guess; if we want to list them all, we have to list them all: 

```suggestion
- `TaskGroup.addTask(UnlessCancelled)`
```

(or, just say "all")